packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -859,15 +859,22 @@ class IntelliJValidatorOnMac extends IntelliJValidator {
 
   @override
   String get pluginsPath {
-    final List<String> split = version.split('.');
-    final String major = split[0];
-    final String minor = split[1];
-    return globals.fs.path.join(
-      globals.fsUtils.homeDirPath,
-      'Library',
-      'Application Support',
-      '$id$major.$minor',
-    );
+    final String plistFile = globals.fs.path.join(installPath, 'Contents', 'Info.plist');
```

Comment:
Cache the result in a field to avoid recomputing every time. See the `version` getter above.